cmake/CMakeLists.txt

```diff
@@ -917,28 +927,126 @@ endif()
 
 if (onnxruntime_ENABLE_TRAINING)
   add_compile_definitions(ENABLE_TRAINING)
+
+  if (UNIX)
+    # Find MPI
+    find_path(MPI_INCLUDE_DIR
+      NAMES mpi.h
+      HINTS
+      ${onnxruntime_MPI_HOME}/include
+      /bert_ort/openmpi/include)
+
+    set(MPI_LIBNAME "mpi")
+
+    find_library(MPI_LIBRARY
+      NAMES ${MPI_LIBNAME}
+      HINTS
+      ${onnxruntime_MPI_HOME}/lib
+      /bert_ort/openmpi/lib)
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(MPI DEFAULT_MSG MPI_INCLUDE_DIR MPI_LIBRARY)
+
+    if (MPI_FOUND)
+      set(MPI_HEADER_FILE "${MPI_INCLUDE_DIR}/mpi.h")
+      message( STATUS "Determining MPI version from the header file: ${MPI_HEADER_FILE}" )
+      file (STRINGS ${MPI_HEADER_FILE} MPI_MAJOR_VERSION_DEFINED
```

Comment:
I think horovod used to use this logic to get version which proved to be unreliable since there are multiple MPI implementations out there. We can simply use something like:
execute_process(COMMAND mpirun --version RESULT_VARIABLE mpirun_output)
message("mpi version='${mpirun_output}'")

Reply:
ok, will update the PR.